### PR TITLE
compare Non-nullable guid

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

*This pull request fixes issue #https://github.com/OData/WebApi/issues/2307.*

### Description

*We have the code to compare non-nullable guid and compare them as "null".

Guid is value type, how can we compare to null?

This PR is to fix that.

### Checklist (Uncheck if it is not completed)

- [] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs [hunk 1]
```diff
@@ -1103,27 +1100,6 @@ internal static Expression ConvertNull(Expression expression, Type type)
...
-            {
-                return firstValue.CompareTo(secondValue);
-            }
-
-            if (secondValue != null)
-            {
-                return (-1) * secondValue.CompareTo(firstValue);
-            }
-
-            return 0;
```
Comment (mikepizzo) [inline src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs:1124]: This is technically a breaking change to a public interface, although I doubt it is being used we could keep it here and just not use it when building the expression (and remove in 8.0).

I'm also fine removing if we are pretty confident it's not being used externally.

**Comment by mikepizzo** [comment]
See my one comment -- removing GuidCompare is technically a breaking change. 

It seems highly unlikely that anyone is using it, but if we think this could break anyone we could keep it and just not use it when building an expression.

**Review by mikepizzo — APPROVED** [review]
:shipit:

File: src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs (hunk 1 above)
Comment (xuzhg) [reply src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs:1124]: I do think there's less customer who relies on this method.  It's "bad" method and we should fix before it broadcasts to more customer. Besides, it's a "easy" task at customer side to re-create a similar GUID compare method. So, in my opinion, let's remove it.

**Pull request merged**